shared/data/src/main/java/com/ivy/data/repository/mapper/TransactionMapper.kt

```diff
@@ -0,0 +1,214 @@
+            val zoneId = ZoneId.systemDefault()
+            val metadata = TransactionMetadata(
+                recurringRuleId = recurringRuleId,
+                loanId = loanId,
+                loanRecordId = loanRecordId
+            )
+
+            val value = Value(
+                amount = PositiveDouble.from(amount).bind(),
+                asset = accountAssetCode ?: AssetCode.from("").bind()
+            )
+
+            Income(
+                id = TransactionId(id),
+                title = title?.let { NotBlankTrimmedString.from(it).bind() },
+                description = description?.let { NotBlankTrimmedString.from(it).bind() },
+                category = categoryId?.let { CategoryId(it) },
+                time = dateTime?.atZone(zoneId)?.toInstant() ?: Instant.now(),
+                settled = true,
+                metadata = metadata,
+                lastUpdated = dateTime?.atZone(zoneId)?.toInstant() ?: Instant.now(),
+                removed = isDeleted,
+                value = value,
+                account = AccountId(accountId)
+            )
+        }
+    }
+
+    private fun TransactionEntity.toExpenseModel(accountAssetCode: AssetCode?): Either<String, Expense> {
+        return either {
+            val zoneId = ZoneId.systemDefault()
+            val metadata = TransactionMetadata(
+                recurringRuleId = recurringRuleId,
+                loanId = loanId,
+                loanRecordId = loanRecordId
+            )
+
+            val value = Value(
+                amount = PositiveDouble.from(amount).bind(),
+                asset = accountAssetCode ?: AssetCode.from("").bind()
+            )
+
+            Expense(
+                id = TransactionId(id),
+                title = title?.let { NotBlankTrimmedString.from(it).bind() },
+                description = description?.let { NotBlankTrimmedString.from(it).bind() },
+                category = categoryId?.let { CategoryId(it) },
+                time = dateTime?.atZone(zoneId)?.toInstant() ?: Instant.now(),
+                settled = true,
+                metadata = metadata,
+                lastUpdated = dateTime?.atZone(zoneId)?.toInstant() ?: Instant.now(),
+                removed = isDeleted,
+                value = value,
+                account = AccountId(accountId)
+            )
+        }
+    }
+
+    private fun TransactionEntity.toTransferModel(
+        fromAccountAssetCode: AssetCode?,
+        toAccountAssetCode: AssetCode?
+    ): Either<String, Transfer> {
+        return either {
+            val zoneId = ZoneId.systemDefault()
+            val metadata = TransactionMetadata(
+                recurringRuleId = recurringRuleId,
+                loanId = loanId,
+                loanRecordId = loanRecordId
+            )
+
+            val fromValue = Value(
+                amount = PositiveDouble.from(amount).bind(),
+                asset = fromAccountAssetCode ?: AssetCode.from("").bind()
```

Comment:
`AssetCode.from("")` always results in an error. You can inspiration from my mapper that handles some edge-cases better:

https://github.com/Ivy-Apps/ivy-wallet/blob/data-layer-part-2/shared/data/src/main/java/com/ivy/data/repository/mapper/TransactionMapper.kt